### PR TITLE
Draft `PHP 8.1` min.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

**Comment by terabytesoftw** [comment]
This is a preliminary version, some comments:

- The version of `PHPUNIT` was increased to `9.6`.
- Compatibility patches have been removed.
- The db tests were moved from build to each db ci, so they won't be tested twice and they will run faster.
- The code was not touched, and only changed in tests, depreciated assertions.
- Added tests for each db controller, for `8.0`, `8.1`, `8.2`.
- The tests for PHP `8.0` for some reason keep getting the segment fault error, so the ci has `continue-error: true`, but the tests all run correctly.
- The `Object::class` was renamed to `ObjectDeprecated::class` to avoid the error when executing the tests.

**Comment by samdark** [comment]
> The tests for PHP 8.0 for some reason keep getting the segment fault error, so the ci has continue-error: true, but the tests all run correctly.

I think we can skip 8.0 and go with 8.1 as minimal version.

> The version of PHPUNIT was increased to 9.6.

Should it be 10?

File: .github/workflows/ci-node.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  test:
+    name: NPM 6 on ubuntu-latest
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout.
+        uses: actions/checkout@v3
+
+      - name: Install PHP.
```
Comment (samdark) [inline .github/workflows/ci-node.yml:23]: Do we need PHP for these?

**Comment by terabytesoftw** [comment]
image

- Code coverage went up a bit, you have to improve the tests and increase it, but it will be in another PR.
- Build-node tests need to be fixed, we need help here.

File: .github/workflows/ci-node.yml (hunk 1 above)
Comment (samdark) [reply .github/workflows/ci-node.yml:24]: I mean these are node tests.

Comment (terabytesoftw) [reply .github/workflows/ci-node.yml]: done.

File: tests/framework/di/InstanceTest.php [hunk 2]
```diff
@@ -58,8 +58,8 @@ public function testEnsure()
     public function testEnsure_NonExistingComponentException()
     {
         $container = new Container();
-        $this->expectException('yii\base\InvalidConfigException');
-        $this->expectExceptionMessageRegExp('/^Failed to instantiate component or class/i');
+        $this->expectException(\yii\base\InvalidConfigException::class);
+        //$this->expectExceptionMessageRegExp('/^Failed to instantiate component or class/i');
```
Comment (samdark) [inline tests/framework/di/InstanceTest.php:62]: ?

File: tests/framework/di/InstanceTest.php [hunk 3]
```diff
@@ -69,8 +69,8 @@ public function testEnsure_NonExistingComponentException()
     public function testEnsure_NonExistingClassException()
     {
         $container = new Container();
-        $this->expectException('yii\base\InvalidConfigException');
-        $this->expectExceptionMessageRegExp('/^Failed to instantiate component or class/i');
+        $this->expectException(\yii\base\InvalidConfigException::class);
+        //$this->expectExceptionMessageRegExp('/^Failed to instantiate component or class/i');
```
Comment (samdark) [inline tests/framework/di/InstanceTest.php:73]: ?

File: tests/framework/web/ErrorActionTest.php [hunk 4]
```diff
@@ -104,20 +104,20 @@ public function testDefaultView()
...
-        $this->expectException('yii\base\ViewNotFoundException');
+        $this->expectException(\yii\base\ViewNotFoundException::class);
 
         $this->getController([
             'layout' => 'non-existing',
         ])->runAction('error');
 
         $ds = preg_quote(DIRECTORY_SEPARATOR, '\\');
-        $this->expectExceptionMessageRegExp('#The view file does not exist: .*?views' . $ds . 'layouts' . $ds . 'non-existing.php#');
+        //$this->expectExceptionMessageRegExp('#The view file does not exist: .*?views' . $ds . 'layouts' . $ds . 'non-existing.php#');
```
Comment (samdark) [inline tests/framework/web/ErrorActionTest.php:121]: ?

File: tests/framework/web/ErrorActionTest.php [hunk 5]
```diff
@@ -104,20 +104,20 @@ public function testDefaultView()
         // Unset view name. Class should try to load view that matches action name by default
         $action->view = null;
         $ds = preg_quote(DIRECTORY_SEPARATOR, '\\');
-        $this->expectException('yii\base\ViewNotFoundException');
-        $this->expectExceptionMessageRegExp('#The view file does not exist: .*?views' . $ds . 'test' . $ds . 'error.php#');
+        $this->expectException(\yii\base\ViewNotFoundException::class);
+        //$this->expectExceptionMessageRegExp('#The view file does not exist: .*?views' . $ds . 'test' . $ds . 'error.php#');
```
Comment (samdark) [inline tests/framework/web/ErrorActionTest.php:108]: ?

**Review by samdark — APPROVED** [review]

File: tests/framework/di/InstanceTest.php (hunk 2 above)
Comment (terabytesoftw) [reply tests/framework/di/InstanceTest.php]: Fixed.

File: tests/framework/di/InstanceTest.php (hunk 3 above)
Comment (terabytesoftw) [reply tests/framework/di/InstanceTest.php]: Fixed.

File: tests/framework/web/ErrorActionTest.php (hunk 4 above)
Comment (terabytesoftw) [reply tests/framework/web/ErrorActionTest.php]: Fixed.

File: tests/framework/web/ErrorActionTest.php (hunk 5 above)
Comment (terabytesoftw) [reply tests/framework/web/ErrorActionTest.php]: Fixed.

**Comment by samdark** [comment]
👍

**Comment by rhertogh** [comment]
Nice 👍

**Pull request merged**